client/landing/gutenboarding/path.ts

```diff
@@ -16,7 +16,7 @@ export const Step = {
 export const langs: string[] = getLanguageSlugs();
 export const steps = Object.keys( Step ).map( key => Step[ key as keyof typeof Step ] );
 
-export const path = `/:step(${ steps.join( '|' ) })?/:lang(${ langs.join( '|' ) })?`;
+export const path = `/:step(${ steps.join( '|' ) })/:lang(${ langs.join( '|' ) })?`;
```

Comment:
@sirreal I was having trouble with this one when the `?` was removed, I believe it is StepType related, probably related to this https://github.com/Automattic/wp-calypso/pull/39312/commits/2880b7b0947acc48521b62bafe8d078081015c4d.

Let me run this again without `?` and try to recall what was the issue.

Reply:
Because of the nature of `<Link>` where it fallbacks to regular `<Button>` when `props.to` is `undefined`, the `props.to` param in `<Link>` accepts all values from StepType and the `undefined` value.

And `usePath()` hook needs to be called outside of conditionals, meaning `usePath(undefined)` will definitely happen when `props.to` is `undefined`. That's also why I added `undefined` in the acceptable params here https://github.com/Automattic/wp-calypso/commit/2880b7b0947acc48521b62bafe8d078081015c4d#diff-e5cab76d1da478606aa476fb05897d00L30.

And because of that, the `path` needed to have the optional `?` for step. To maintain the stance that step shouldn't be an optional parameter in `path`, I think one way around it is not to call `generatePath` when `makePath/usePath` receives an undefined step.

Reply:
So this is how `makePath/usePath` looks like if step is not an optional `?` parameter.

```jsx
export function makePath( step: StepType | undefined, lang?: string ) {
    // When step is undefined (coming from <Link>).
	if (!step) {
		return '/' + (lang && langs.includes( lang ) ? lang : '');
	}

	return generatePath( path, {
		step,
		...( lang && langs.includes( lang ) && { lang } ),
	} );
}
```

I think the `path` route definition should also accept `/fr` as a valid definition?

Reply:
Also can be seen here:
https://github.com/Automattic/wp-calypso/commit/cf6ec5b8d99f0931a62d1d50af1a595e777c368d#diff-e5cab76d1da478606aa476fb05897d00R27-R29

Reply:
> Because of the nature of `<Link>` where it fallbacks to regular `<Button>` when `props.to` is `undefined`, the `props.to` param in `<Link>` accepts all values from StepType and the `undefined` value.
> 
> And `usePath()` hook needs to be called outside of conditionals, meaning `usePath(undefined)` will definitely happen when `props.to` is `undefined`. That's also why I added `undefined` in the acceptable params here https://github.com/Automattic/wp-calypso/commit/2880b7b0947acc48521b62bafe8d078081015c4d#diff-e5cab76d1da478606aa476fb05897d00L30.

I left a [comment](https://github.com/Automattic/wp-calypso/pull/39312/files#r380180475) on the changes to `Link`, arguing that `Link` should be ignorant of `usePath`.